server/test/services/openapi/v2/Swagger2SchemaGeneratorTest.java

```diff
@@ -0,0 +1,363 @@
+            .appendLine("              description: A link to the next page of responses")
+            .appendLine("          schema:")
+            .appendLine("            $ref: \"#/definitions/result\"")
+            .appendLine("        \"400\":")
+            .appendLine(
+                "          description: Returned if any request parameters fail validation.")
+            .appendLine("        \"401\":")
+            .appendLine(
+                "          description: Returned if the API key is invalid or does not have access"
+                    + " to the program.")
+            .appendLine("      tags:")
+            .appendLine("        - programs")
+            .appendLine("definitions:")
+            .appendLine("  result:")
+            .appendLine("    type: object")
+            .appendLine("    properties:")
+            .appendLine("      payload:")
+            .appendLine("        type: array")
+            .appendLine("        items:")
+            .appendLine("          type: object")
+            .appendLine("          properties:")
+            .appendLine("            applicant_id:")
+            .appendLine("              type: integer")
+            .appendLine("              format: int32")
+            .appendLine("            application:")
+            .appendLine("              type: object")
+            .appendLine("              properties:")
+            .appendLine("                name:")
+            .appendLine("                  type: object")
+            .appendLine("                  properties:")
+            .appendLine("                    question_type:")
+            .appendLine("                      type: string")
+            .appendLine("                    first_name:")
+            .appendLine("                      type: string")
+            .appendLine("                      x-nullable: true")
+            .appendLine("                    last_name:")
+            .appendLine("                      type: string")
+            .appendLine("                      x-nullable: true")
+            .appendLine("                    middle_name:")
+            .appendLine("                      type: string")
+            .appendLine("                      x-nullable: true")
+            .appendLine("                sample_address_question:")
+            .appendLine("                  type: object")
+            .appendLine("                  properties:")
+            .appendLine("                    question_type:")
+            .appendLine("                      type: string")
+            .appendLine("                    city:")
+            .appendLine("                      type: string")
+            .appendLine("                      x-nullable: true")
+            .appendLine("                    corrected:")
+            .appendLine("                      type: string")
+            .appendLine("                      x-nullable: true")
```

Comment:
**suggestion**: This is an enum, but I don't think we have an enum type yet in this serializer.

One thought around enums is that I've read that the default swagger generated consuming code can sometimes assume that the current enum values are all that will ever be present, rather then being built to handle future unknown values, so maybe leaving it as a string is better, to force a human to think about it?

Reply:
See notes on it here https://github.com/civiform/civiform/pull/7051#discussion_r1876976400